lib/autoupdate/rollout/strategy_haltonerror_test.go

```diff
@@ -500,7 +500,7 @@ func Test_progressGroupsHaltOnError(t *testing.T) {
 				State:     0,
 				StartTime: tt.rolloutStartTime,
 			}
-			err := strategy.progressRollout(ctx, status, clock.Now())
+			err := strategy.progressRollout(ctx, nil, status, clock.Now())
```

Comment:
Is calling progressRollout with a nil spec expected to happen in practice? Should this error?

Reply:
There's a single "real" call to this interface and it always sets the spec. Strategies can expect the spec to be always set.

The nil spec in the test is me taking a shortcut because I know that this specific strategy doesn't rely on the spec.